### PR TITLE
Link across database

### LINKED ISSUE #1018: Add support for links across databases.
### The problem faced currently?
Currently we need to use on the fly joins and manual mutations for working with linked tables across multiple databases

### How can we solve it?
There should be a `db` argument to the `@link` directive which sets the dbAlias field in the `LinkTable Properties`. This alone will take care of linked queries.
We also have to update the mutation  mechanism.

>If you want this feature to be implemented, give it a thumbs up reaction, so that we can determine which features are important to you.
>👍

### PR DESCRIPTION
**Pull request by prithvi081099**
Fixes #1018

### CODE REVIEW

**Comment by sharadregoti** [comment]
@YourTechBud how can we test this pr ?
As open source version doesnt allow to have multiple databases ?

**Comment by YourTechBud** [comment]
@sharadregoti lets just merge it for now. We'll test it later

**Comment by YourTechBud** [comment]
@prithvi081099 run `go mod tidy` and push it again

**Comment by sharadregoti** [comment]
@YourTechBud does this code work for both query & mutations ?

**Comment by YourTechBud** [comment]
Yes

File: gateway/utils/graphql/mutation.go [hunk 1]
```diff
@@ -116,7 +118,9 @@ func (graph *Module) handleMutation(ctx context.Context, node ast.Node, token st
 
 		// Add the request to the number of requests available for that database
 		r = append(r, generatedRequests...)
-		reqs[dbAlias] = r
+		for _, v := range r {
```
Comment (YourTechBud) [inline gateway/utils/graphql/mutation.go:121]: Range over `generatedRequests`. We don't need `r`

File: gateway/utils/graphql/mutation.go [hunk 2]
```diff
@@ -32,6 +32,7 @@ func (graph *Module) generateAllReq(ctx context.Context, field *ast.Field, token
 			}
 			result.Type = string(utils.Delete)
 			result.Col = col
+			result.DBAlias, _ = graph.GetDBAlias(field)
```
Comment (YourTechBud) [inline gateway/utils/graphql/mutation.go:35]: Instead of doing this, why dont you just pass the dbAlias as a parameter in this function

**Pull request merged**